public/app/plugins/panel/alertlist/UnifiedAlertList.tsx

```diff
@@ -337,7 +337,12 @@ function filterRules(props: PanelProps<UnifiedAlertListOptions>, rules: Combined
           alertingRule.alerts ?? []
         )
       : [];
-    if (filteredAlerts.length) {
+    if (
+      filteredAlerts.length ||
+      (alertingRule?.state === PromAlertingRuleState.Inactive &&
+        options.showInactiveAlerts &&
+        !options.alertInstanceLabelFilter.length)
```

Comment:
Q: I'm a bit confused why we are also checking the `alertInstanceLabelFilter` here 🤔

Reply:
This is to basically give any instance label filters the user gave precedence over this setting. I tried to convey it in the FE for the user (below), but should have commented here too.
> If an Alert instance label filter is configured, inactive alert rules will be hidden

Most of the filters act on the alert rule itself; Alert name, datasource, folder. The one filter that acts on alert instances is `alertInstanceLabelFilter`, but seeing these inactive rules have zero instances, the filter had no effect. Here I check if the user provided any label filters, and if they did, hide the inactive alert rules.